partition/__main__.py

```diff
@@ -69,41 +85,50 @@ def has_duplicates(mapping):
         return False
 
 
+def write_relpaths(relpaths=[], file=sys.stdout):
```

Comment:
It is a Python best practice not to use mutable values (like a list or dictionary literal) as the default arguments to a function. Since the function defaults are only evaluated once (when the function is defined), all calls to the function that rely on the default will use the same object. Probably not a big deal in this case, since you are only calling this function once anyway, but it is a good habit to get into :-)

For more info, see: <https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html>

Rewrite suggestion:

```python
def write_relpaths(relpaths=None, file=sys.stdout):
    if relpaths is None:
        relpaths = []
    ...
```
This creates a new list object each time the function is called without a `relpaths` argument.